### ISSUE TITLE
Feature request: Safety valves for airpumps/canisters

### ISSUE THREAD

**Issue by FudgePucker** [d3athrow/vgstation13#12204]
I actually got this idea while I was sitting on the toilet thinking about how new players might be scared off initially from engineering jobs and atmospherics in general.
Seriously this is the most genius feature request I have ever made.
I am surprised no one has thought of this before.

Lets say you want to fill a plasma tank with some plasma, you put the tank into the canister and open the valve to fill it. But opening the valve is the same input for releasing the canister into the air. So a new player will end up forgetting to close the valve before removing the tank, causing the contents of the canister to drain into the air, causing a plasma flood.
The most genus way to prevent this is a safety valve, it prevents the canister from leaking gas, but doesn't block it from filling tanks. Meaning if you leave the canister open when filling a tank and remove the tank without closing it, it doesn't spill out. Want to repressure an area using a canister of air? Just disable the safety valve and open it.

This would also be a great QoL update for engineers, because when filling plasma tanks for collectors you don't have to keep turning the pump or canister on and off every time you put in a tank, you just pop in a tank, wait for it to update to fill, and remove it, rinse and repeat.

We have all been the idiot who forgot to turn off the canister or pump when removing tanks, and we have all seen new players do it as well. This would be the single most greatest QoL update ever, preventing rookie mistakes, and also making it less tedious to fill multiple tanks.
![voxarepox](https://cloud.githubusercontent.com/assets/12202731/19370460/f82bcbae-9179-11e6-888c-e41fe2a5d0ca.png)

**Comment by clusterfack** [comment]
Lol? Im 99.99% this is completely intended

**Comment by FudgePucker** [comment]
Is it really though?
Why is having plasma floods from a simple mistake a intended feature?
Why is having to deal with three nano ui inputs to just fill a single tank a feature?
Do you have a better argument other than its intended to be tedious and punishing to just fill a tank?

**Comment by Exxion** [comment]
Yes, it's intended, but the important question is if it _should_ work that way

**Comment by AlphaBuffalo** [comment]
I vote to remove it, some intended features are not worth keeping

**Comment by clusterfack** [comment]
Because its hilarious

**Comment by Aurx** [comment]
Being able to flood via a canister is a feature that shouldn't be removed.

**Comment by Shadowmech88** [comment]
I think that players having the capability to unintentionally majorly fuck up adds to the spice of any role.

In fact if there's anything in Engineering that needs a safety switch to prevent newbies from fucking it up and causing damage, it's the singulo.

**Comment by BobdaBiscuit** [comment]
I was thinking about a similar feature but I do think having people able to accidentally flood is hilarious and adds to rounds to an extent.

**Comment by clusterfack** [comment]
Ss13 is a disaster simulator we should be lookibg for more ways for things to go hilariously wrong not removing them.

**Comment by Kurfursten** [comment]
Yeah I think I kind of like leaks being a thing. Engineering has always been a department with a sort of beautiful terror to it, and becoming a smooth operator just feels more rewarding because of that. There are tons of ways engineers can die because they fucked up their job.

> Touched the containment field
> Forgot to wear glasses, stunned too close to singulo
> Forgot to put on gloves
> Forgot to wear glasses, freaked out over hallucinations from SME
> Fucked up proper containment of the singulo/SME/AME
> Open door without magboots on, slam into wall until dead of bloodloss
> Worked on plasmaflood in a station engineer suit
> Worked on radiation zone in an atmos suit

Engineering is actually quite a boring job without the high stakes. Set up an engine and do a little routine maintenance.

I normally hesitate to use a slippery slope/snowball kind of argument, but adding safety valves to plasma tanks seems like the first step toward adding an emergency shutdown to the radio transmitter if someone is in range or keeping the supermatter shard in a transparent box so that you can use it for an engine with no risk of bumping it for instant death.

**Comment by FudgePucker** [comment]
Its funny is not a good enough reason for this shit.
Hahaha that new player will never play engineer again because he is scared of getting banned if he messes up in engineering again, HOW FUCKING HILARIOUS.
hahaha you messed up your inputs on the shitty nanoui, so now you have to spend thirty minutes in medbay getting fixed up from the toxins damage and eye damage, delaying the engine set up, and everyone will mock you.

Also seriously, are you guys really defending accidental plasmafloods caused by nanoui?

Also like I said before, it would be a HUGE QoL update for engineers. It would make filling multiple plasma tanks for the collectors so much less tedious. One nanoui input vs three for every tank. Seriously, are you guys really defending the tedioum of filling tanks for radiation collectors?

@Kurfursten none of the things you listed that you can fuck up as an engineer involves dealing with the shitty laggy nanoui.

@Aurx wouldn't remove it, just add another step. Also who doesn't just spam welding tool to burst it anyways?

**Comment by clusterfack** [comment]
If people are getting banned for this its an admin problem not a code problem. Admins are too quick to ban for engineering mistakes.

**Comment by FudgePucker** [comment]
I didn't say that they would get banned, the fear of getting in trouble for fucking up is more than enough.
Its a mixture of fear and shame that would make people never try engineering after a fuck up like that.

I mean come on, I really want this for the QoL increase because fuck filling tanks, and fuck nanoui. I know deep down inside you also want to just quickly be able to pop in tanks and take them out with out having to deal with turning the canister on and off and on and off and on and off and on and off and on and off and on and off.

The difficulty for engineering shouldn't come from having to deal with a laggy shitty UI of a canister.

**Comment by Kurfursten** [comment]
@FudgePucker Filling tanks for rad collectors is only tedious because the SME room starts with way too many but they're impossible for any mapper to ever remove because muh numbers.

A buildable tank auto-fill machine that you could hook up to a plasma line would be neat though. Give mechanics something to do that's actually related to engineering and give atmos techs something to harass scientists/mechanics for.

**Comment by icantthinkofanameritenow** [comment]
The canister may have laggy shitty UI, but it's not like that would make it an exception to the rule in this game. 

If you want more people to play as engineers, get them to learn it's okay to fuck up. Everyone does at some point, and when presented with the consequences of carelessness they will learn not to make that mistake again. We don't need to baby the newbies, we need to be even harder on them.

**Comment by FudgePucker** [comment]
@Kurfursten 

> This feature would be too convenient to have round start.
> I KNOW LETS ADD IT TO RESEARCH

GOLLY, IF THERE WAS ONLY A REASON WHY SCIENCE AS A DEPARTMENT IS SO OVER POWERED. I HAVE NO FUCKING IDEA WHY SCIENCE WOULD HAVE SO MUCH SHIT OVER OTHER DEPARTMENTS.

No really, your fucking argument against adding this feature to canisters is to make it a machine locked away behind research? Your solution is to lock it behind that fucking feature dump?

Now toxins can mass produce bombs easier, and engineering never sees the machine.

@icantthinkofanameritenow 
You are ignoring another reason why people don't like to play engineer. Its a tedious boring thankless job that someone has to do every round. Boy setting up the engine every round just to wait for the shuttle to be called as soon as I am done sure is great, I am not getting burned out and hating this game every round I play. Every time its nukeops/wizard/roundstart instant shuttle call, someone sets engineer job preference to off. Its like fucking groundhogs day. Sure you could try setting up some fancy engine, but you can't really do that because you are on a constant timer before the power for the station shuts off. This would remove some of that tediousness. Station can manage if there is no one in cargo, station can manage if there's no one in service, station can manage if there's no one in science, station needs someone to set up the engine.

All this does is help remove some of that tedium of setting up the engine.

**Comment by Kurfursten** [comment]
You make a valid argument, no one should get protection from leaks at all. Calm down tho.

**Comment by icantthinkofanameritenow** [comment]
@FudgePucker

Three extra minutes isn't tedium, and if you can't figure out how to make your own fun as an engine perhaps you shouldn't be playing one in the first place.

Set up the fancy engine after the easy one, that's what the AME is for. There's even the PACMAN which doesn't even need that if you're really lazy. 

> Boy setting up the engine every round just to wait for the shuttle to be called as soon as I am done sure is great, I am not getting burned out and hating this game every round I play.

Ah, that's the problem. You just really, REALLY need a break from the game.and don't want to admit it. Seriously, every request you make is equal parts bitterness and "make the game.easier for me, I want everything handed to me on a silver platter".

Fuck the newbies, they can learn by doing and if the consequences kill them it's one less mistake they'll make in the future.

**Comment by FudgePucker** [comment]
@icantthinkofanameritenow 
There are still plenty of other ways to be killed or fuck everything up other than misclicking on a shitty UI.
Last SME I set up I was trying to speed run the set up so I could make a R-UST where the singulo is, but I forgot to link the scrubbers to the cooling loop, causing it to go super nova, I died trying to space the shard. Also I have only played like two rounds last week, two rounds the entire month, I am in the middle of a long ass break.

All I am saying is that the difficulty from engineering shouldn't come from shitty canister UI.
A newbie can still mess up plenty when setting up an engine.

Also AME is against my religion. I have set it up one or two times tops, one of those times being in cargo as a miner because the QM wanted to set up one.
![basedontrueevents](https://cloud.githubusercontent.com/assets/12202731/19411898/12119cf6-92d8-11e6-8301-f92fc3e462e6.png)
